manifests/charts/ztunnel/templates/daemonset.yaml

```diff
@@ -1,3 +1,5 @@
+{{- $mergedEnv := merge dict (.Values.ztunnel.env | default dict) (.Values.meshConfig.defaultConfig.proxyMetadata | default dict) }}
```

Comment:
This isn't great.. it means we would override proxyConfig which should probably take precedence.

Reply:
A good example of why we should separate proxyconfig for ztunnel completely.